docs/components/mate/creating-extensions.rst

```diff
@@ -0,0 +1,268 @@
+Creating MCP Extensions
+=======================
+
+MCP extensions are Composer packages that declare themselves using a specific configuration
+in ``composer.json``, similar to PHPStan extensions.
+
+Quick Start
+-----------
+
+1. Configure composer.json
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: json
+
+    {
+        "name": "vendor/my-extension",
+        "type": "library",
```

Comment:
we could also go with a specific type here, similar to phpstan:
```suggestion
        "type": "mate-extension",
```

Reply:
@chr-hertel i also think it could make sense to have the tools also included in a "bigger" repo and so i would also ignore the package type